src/sleeper.rs

```diff
@@ -0,0 +1,14 @@
+use std::time::Duration;
+
+pub trait Sleeper {
+    fn sleep(duration: Duration) -> impl std::future::Future<Output = ()> + Send;
+}
+
+pub struct DefaultSleeper;
```

Comment:
We can derive `Debug` and `Clone` for this so that `AsyncClient` can do the same. Items in this module should have doc comments also.